components/PaymentOutcome/index.tsx

```diff
@@ -16,13 +17,30 @@ interface Props {
 
 function PaymentOutcome({ paymentRequest, paymentAmount, dispatch }: Props) {
   const router = useRouter()
-  const { amount, unit, sats } = router.query
+  const { amount, unit, sats, username, memo } = router.query
   const { satsToUsd } = useSatPrice()
+  const [isPrint, setIsPrint] = React.useState(false)
+  const [show, setShow] = React.useState(false)
 
   if (!paymentRequest) {
     return null
   }
 
+  const printReceipt = () => {
+    setIsPrint(true)
+    setShow(true)
+    setTimeout(() => {
```

Comment:
Why do we need this timeout? Why can't we call the if statement directly instead of wrapping it in a timeout?

Reply:
Hi Theo. Thanks for the comment. I have pushed an update that removes the need for a timeout. Looking forward to your feedback. Kind regards.